### PR TITLE
Add backdrop for dropdown

### LINKED ISSUE #581: All pages that have dropdown selectors should have an overlay background
Please check all pages that contain dropdown:

We should add an overlay so that when the overlay is tapped, we close the dropdown
Currently the only way to close a dropdown is by selecting an item

### PR DESCRIPTION
**Pull request by JustinBeBoy**
This PR resolve #581 with detail:
- Add backdrop for dropdown
- Implement pages which using dropdown

### CODE REVIEW

File: ui/page/tickets/list_page.go [hunk 1]
```diff
@@ -121,6 +121,9 @@ func (pg *ListPage) Layout(gtx layout.Context) layout.Dimensions {
 							})
 						})
 					}),
+					layout.Expanded(pg.orderDropDown.BackDrop),
+					layout.Expanded(pg.ticketTypeDropDown.BackDrop),
+					layout.Expanded(pg.walletDropDown.BackDrop),
```
Comment (Sirmorrison) [inline ui/page/tickets/list_page.go]: This should be added to the `dropdown` widget as part of the layout method.

File: ui/page/transactions_page.go [hunk 2]
```diff
@@ -162,6 +162,9 @@ func (pg *TransactionsPage) Layout(gtx layout.Context) layout.Dimensions {
 					})
 				})
 			}),
+			layout.Expanded(pg.txTypeDropDown.BackDrop),
+			layout.Expanded(pg.orderDropDown.BackDrop),
+			layout.Expanded(pg.walletDropDown.BackDrop),
```
Comment (Sirmorrison) [inline ui/page/transactions_page.go]: This should be part of the `dropdown` widget not the page the dropdown is used.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Review by Sirmorrison — DISMISSED** [review]

**Comment by dreacot** [comment]
@JustinBeBoy This hasn't been implemented in the ticket activity page

**Comment by Sirmorrison** [comment]
> @JustinBeBoy This hasn't been implemented in the ticket activity page

This is now resolved.

**Review by dreacot — DISMISSED** [review]

File: ui/page/tickets/activity_page.go [hunk 3]
```diff
@@ -103,6 +105,13 @@ func (pg *ActivityPage) Layout(gtx layout.Context) layout.Dimensions {
 							})
 						})
 					}),
+					layout.Expanded(func(gtx C) D {
+						if pg.orderDropDown.IsOpen() || pg.ticketTypeDropDown.IsOpen() || pg.walletDropDown.IsOpen() {
+							return pg.backdrop.Layout(gtx)
```
Comment (beansgum) [inline ui/page/tickets/activity_page.go]: This implementation requires each page to have a backdrop, the backdrop should be handled by the dropdown instead.

**Review by beansgum — CHANGES_REQUESTED** [review]

File: ui/page/transactions_page.go [hunk 4]
```diff
@@ -162,38 +162,24 @@ func (pg *TransactionsPage) Layout(gtx layout.Context) layout.Dimensions {
 					})
 				})
 			}),
-			layout.Stacked(pg.dropDowns),
+			layout.Expanded(func(gtx C) D {
+				return pg.walletDropDown.Layout(gtx, 0)
+			}),
+			layout.Expanded(func(gtx C) D {
+				post := float32(gtx.Constraints.Max.X - 315)
```
Comment (Sirmorrison) [inline ui/page/transactions_page.go]: rename all occurrence of `post` to `position` or `pos`

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
image

 on the ticket page this is what the dropdowns look like at the far right.

Same for ticket activity page.

image

Same for transactions page.
image

**Comment by JustinBeBoy** [comment]
> image
> 
> on the ticket page this is what the dropdowns look like at the far right.
> 
> Same for ticket activity page.
> 
> image
> 
> Same for transactions page.
> image

I cant reproduce it in computer, can you tell me steps to reproduce it

**Comment by beansgum** [comment]
@Sirmorrison I can't also reproduce the ui bug.

**Comment by Sirmorrison** [comment]
> @Sirmorrison I can't also reproduce the ui bug.

Let me try from my mac machine it was noticed on windows

this is what i get on my macbook

transaction page

image

ticket page

image

File: ui/page/tickets/activity_page.go [hunk 5]
```diff
@@ -103,8 +101,16 @@ func (pg *ActivityPage) Layout(gtx layout.Context) layout.Dimensions {
...
+					layout.Expanded(func(gtx C) D {
+						return pg.walletDropDown.Layout(gtx, 0)
+					}),
+					layout.Expanded(func(gtx C) D {
+						post := float32(gtx.Constraints.Max.X - 150)
+						return pg.orderDropDown.Layout(gtx, post)
+					}),
+					layout.Expanded(func(gtx C) D {
+						post := float32(gtx.Constraints.Max.X - 315)
+						return pg.ticketTypeDropDown.Layout(gtx, post)
```
Comment (Sirmorrison) [inline ui/page/tickets/activity_page.go]: So doing troubleshooting, this seems to be where the issue is coming from.

Changing the values as suggested below, give me the screenshots below.
image

```suggestion
					layout.Expanded(func(gtx C) D {
						return pg.walletDropDown.Layout(gtx, 0)
					}),
					layout.Expanded(func(gtx C) D {
						post := float32(gtx.Constraints.Max.X - 450)
						return pg.orderDropDown.Layout(gtx, post)
					}),
					layout.Expanded(func(gtx C) D {
						post := float32(gtx.Constraints.Max.X - 300)
						return pg.ticketTypeDropDown.Layout(gtx, post)
```

I think the issue here has to do with the screen size of the machine. you will need to use `gtx.Px()` for positioning and scaling.

**Review by Sirmorrison — CHANGES_REQUESTED** [review]

**Comment by Sirmorrison** [comment]
image

The space between the dropdown is too much... See the red section on the screenshot.

**Review by Sirmorrison — APPROVED** [review]
PR looks good.

**Review by beansgum — APPROVED** [review]

**Pull request merged**